File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(l,r):
    # 각 수당 약수의 개수 구하기
    sum = 0
    for i in range(l,r+1):
        # 각 수 제곱근까지 나누어 떨어지는지를 확인
        cnt = 0
        for j in range(1, int(i**(0.5) ) + 1 ):
            if i % j == 0:
                if i / j == j: # 제곱근일 경우
                    cnt += 1
                else:
                    cnt += 2
        #print(cnt)
        if cnt % 2 == 0:
            sum += i
        else:
            sum -= i
    #print(sum)
    return sum

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public static boolean isAdd(int num){
        // 만약 루트한 것이 정수이면 홀수 -> 빼기
        double num_sqrt = Math.sqrt(num);
        if(num_sqrt == (int)num_sqrt){
            return false;
        }
        return true;
        
    }
    public int solution(int left, int right) {
        /*
        left - right까지 순회하면서 약수 개수 구하기
        */
        int sum = 0;
         for(int i=left; i<=right; i++){
             if(isAdd(i)){
                 sum += i;
             }
             else{
                 sum -= i;
             }
         }
        return sum;
    }
}